parsl/executors/high_throughput/htex.py

```diff
@@ -0,0 +1,392 @@
+            raise ConfigurationError('Multiple storage access schemes are not yet supported')
+        self.working_dir = working_dir
+        self.managed = managed
+        self.engines = []
+        self.tasks = {}
+
+        self.public_ip = public_ip
+        self.worker_ports = worker_ports
+        self.worker_port_range = worker_port_range
+        self.internal_port_range = internal_port_range
+
+        if not launch_cmd:
+            self.launch_cmd = """process_worker_pool.py {debug} -w {tasks_per_node} --task_url={task_url} --result_url={result_url}"""
+
+    def start(self):
+        """ Here we create the Interchange process and connect to it.
+        """
+        self.outgoing_q = zmq_pipes.TasksOutgoing('tcp://127.0.0.1:50055')
+        self.incoming_q = zmq_pipes.ResultsIncoming('tcp://127.0.0.1:50056')
+
+        self.is_alive = True
+
+        self._queue_management_thread = None
+        self._start_queue_management_thread()
+        self._start_local_queue_process()
+
+        logger.debug("Created management thread : %s", self._queue_management_thread)
+
+        if self.provider:
+            debug_opts = "--debug" if self.engine_debug else ""
+            l_cmd = self.launch_cmd.format(debug=debug_opts,
+                                           task_url=self.worker_task_url,
+                                           result_url=self.worker_result_url,
+                                           tasks_per_node=self.provider.tasks_per_node,
+                                           nodes_per_block=self.provider.nodes_per_block)
+            self.launch_cmd = l_cmd
+            logger.debug("Launch command :{}".format(self.launch_cmd))
+
+            self._scaling_enabled = self.provider.scaling_enabled
+            logger.debug("Starting HighThroughputExecutor with provider:\n%s", self.provider)
+            if hasattr(self.provider, 'init_blocks'):
+                try:
+                    for i in range(self.provider.init_blocks):
+                        engine = self.provider.submit(self.launch_cmd, 1)
+                        logger.debug("Launched block: {0}:{1}".format(i, engine))
+                        if not engine:
+                            raise(ScalingFailed(self.provider.label,
+                                                "Attempts to provision nodes via provider has failed"))
+                        self.engines.extend([engine])
+
+                except Exception as e:
+                    logger.error("Scaling out failed: %s" % e)
+                    raise e
+
+        else:
+            self._scaling_enabled = False
+            logger.debug("Starting IpyParallelExecutor with no provider")
+
+    def _queue_management_worker(self):
+        """Listen to the queue for task status messages and handle them.
+
+        Depending on the message, tasks will be updated with results, exceptions,
+        or updates. It expects the following messages:
+
+        .. code:: python
+
```

Comment:
What I was trying to do was leave in the possibility to squash/piggyback some messages. This might need to be reworked based on some pending experiments on how useful batching is for results. It might well be the case that zmq already does that.